libdnf/repo/Repo.cpp

```diff
@@ -822,7 +822,13 @@ std::vector<Key> Repo::Impl::retrieve(const std::string & url)
         close(fd);
     });
 
-    downloadUrl(url.c_str(), fd);
+    try {
+        downloadUrl(url.c_str(), fd);
+    }
+    catch (const LrExceptionWithSourceUrl & e) {
```

Comment:
Is this the only exception that can be thrown here or do we perhaps want to catch all exceptions?

Reply:
I'd say if other exception should appear, we don't want to catch it anyway. This just catches the LrExceptionWithSourceUrl before it gets to the load function (where it would be then catched) in order to change the message.